src/Arcus.BackgroundJobs.KeyVault/Extensions/ServiceBusMessageHandlerCollectionExtensions.cs

```diff
@@ -120,6 +119,104 @@ public static class ServiceBusMessageHandlerCollectionExtensions
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///     Thrown when the <paramref name="subscriptionNamePrefix"/> or <paramref name="serviceBusTopicConnectionStringSecretKey"/> is blank.
+        /// </exception>
+        public static ServiceBusMessageHandlerCollection WithAutoRestartOnRotatedCredentials(
+            this ServiceBusMessageHandlerCollection services,
+            string subscriptionNamePrefix,
+            string serviceBusTopicConnectionStringSecretKey,
+            string messagePumpConnectionStringKey)
+        {
+            return WithAutoRestartOnRotatedCredentials(
+                services, subscriptionNamePrefix, serviceBusTopicConnectionStringSecretKey, messagePumpConnectionStringKey, configureBackgroundJob: null);
+        }
+
+        /// <summary>
+        /// Adds a background job to the <see cref="IServiceCollection"/> to automatically restart a <see cref="AzureServiceBusMessagePump"/> with a specific <see cref="MessageHandlerCollection.JobId"/>
+        /// when the Azure Key Vault secret that holds the Azure Service Bus connection string was updated.
+        /// </summary>
+        /// <remarks>
+        ///     Make sure that the application has the Arcus secret store configured correctly.
+        ///     For on the Arcus secret store: <a href="https://security.arcus-azure.net/features/secret-store" />.
+        /// </remarks>
+        /// <param name="services">The collection of services to add the job to.</param>
+        /// <param name="subscriptionNamePrefix">The name of the Azure Service Bus subscription that will be created to receive <see cref="CloudEvent"/>'s.</param>
+        /// <param name="serviceBusTopicConnectionStringSecretKey">The secret key that points to the Azure Service Bus Topic connection string.</param>
+        /// <param name="messagePumpConnectionStringKey">
+        ///     The secret key where the connection string credentials are located for the target message pump that needs to be auto-restarted.
+        /// </param>
+        /// <param name="configureBackgroundJob">
+        ///     The capability to configure additional options on how the auto-restart Azure Service Bus message pump
+        ///     on rotated Azure Key Vault credentials background job should behave.
+        /// </param>
+        /// <exception cref="ArgumentNullException">
+        ///     Thrown when the <paramref name="services"/> or the searched for <see cref="AzureServiceBusMessagePump"/> based on the given <see cref="MessageHandlerCollection.JobId"/> is <c>null</c>.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///     Thrown when the <paramref name="subscriptionNamePrefix"/> or <paramref name="serviceBusTopicConnectionStringSecretKey"/> is blank.
+        /// </exception>
+        public static ServiceBusMessageHandlerCollection WithAutoRestartOnRotatedCredentials(
```

Comment:
Added this overload without a job ID that simplifies this, as now the job ID is available in the `ServiceBusMessageHandlerCollection` (set by the previously registered message pump, now available in the new major Messaging release). That way, the user does not have to provide it themselves.